enhancements/machine-config/user-data-secret-managed.md

```diff
@@ -0,0 +1,113 @@
+
+### Goals
+
+The MCO owns the lifecycle of the Secret containing the pointer config. The installer is still responsible to create the initial version of the Secret as that is required to support UPI installation by generating the initial ignition configs until the installer provisions the control plane via MAO and not TF. The MCO takes ownership of the Secret containing the stub ignition once the cluster is up and running. When we'll update to the new Ignition version, the MCO can update the Secret.
+
+The final goal for this enhancement is to be able to manage that Secret from the MCO during the ignition migration to the new spec.
+
+If we don't manage that Secret, its content will be on an unsupported version of Ignition that can't provision new machines joining the cluster.
+
+### Non-Goals
+
+To make this simple and to be able to fit this into the release that will bring Ignition v2, we're not considering per-machine Secret(s) as highlighted here https://github.com/openshift/machine-config-operator/issues/683#issuecomment-493193902 and here https://github.com/openshift/machine-config-operator/issues/683#issuecomment-640602493
+
+We don't want to diverge from the current implementation where the Installer still creates the Secret and stub ignition on installation otherwise we'll have to change far more things in order to keep supporting UPI installation that needs ignition configs before starting to install.
+
+## Proposal
+
+The proposal is pretty simple today:
+
+- The installer keeps creating the Secret that holds the stub ignition config (at any given version) under a new name
+- The secret will change name from `<role>-user-data` to `<role>-user-data-managed` which is going to be v3 at a later time before the release
+- The name change is needed to keep supporting old bootimages that still reference the old ignition version
+- The MCO takes over owning that Secret and its content once the cluster is up and running
+- The MCO has a way to manage the lifecycle of the Secret containing the stub ignition
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+The implementation has already started and there are two PRs that provide a working POC:
+
+- https://github.com/openshift/machine-config-operator/pull/1792
+- https://github.com/openshift/installer/pull/3730
+
+The MCO PR simply learns how to create the content of the secret and once it's running, it goes ahead and takes ownership of the secret in the cluster (you can see we're using the new name ending in `-managed`). That allows us to upgrade the MCO to a newer Ignition version and make sure the secret's content is updated as well.
+The installer PR is just a rename of the old user-data secret to something new that the MCO can manage. The main reason to the rename is ability to still have the old user-data secret around during an upgrade.
```

Comment:
why do you need to have the old one around? as you as mco runs it owns it and can put any content it needs to.

Reply:
replied here https://github.com/openshift/enhancements/pull/368#discussion_r440049493